### ISSUE TITLE
Feature selection and engineering 

### ISSUE THREAD

**Issue by Mikeymice** [UBC-MDS/Capital_Bikeshare_Analysis#15]
Aside the rental type we discussed, I am thinking if we should also omit the the column `year` as well. the reason is that it appears to be the most deciding factor for many different trials of the decision tree which may not be desirable for our project and we only have 2 years

![dtree](https://user-images.githubusercontent.com/15175882/48878714-786fe100-edbd-11e8-89af-2405de4c5307.png)

**Comment by Mikeymice** [comment]
Here is the tree with year removed
![dtree](https://user-images.githubusercontent.com/15175882/48878822-fb913700-edbd-11e8-8897-8f76063b258b.png)

**Comment by Mikeymice** [comment]
Also, instead of using the graph to decide the top features. i found that the tree has the attribute [`feature_importances`](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html#sklearn.tree.DecisionTreeClassifier.feature_importances_ ) which list the Gini importance of each features and according to the list, the top 3 are ['atemp', 'hum', 'windspeed'] but i think we should check with TA and Tiffany

**Comment by EvanYathon** [comment]
I was undecided on removing the `year` feature... I thought could be useful in predicting higher ridership due to the fact that the bike share program ridership is declining/growing over time and could be useful for backing up that hypothesis.  But I also see your point that we are mainly wondering about weather conditions effect on ridership.  I'm good with dropping the `year` feature as it seems to be different from our original question's intent.

With regard to the `feature_importances` I think that's a great attribute to use!  I think Tiffany actually mentioned it also in an [issue](https://github.ubc.ca/MDS-2018-19/DSCI_522_dsci-workflows_students/issues/9) so I think we are good to use it!

**Comment by Mikeymice** [comment]
after digging with the cross validation, we might have to include the year for some reason. because without the year the accuracy of the cross validation is pretty bad 
here is with year 

![accuracy_with_year](https://user-images.githubusercontent.com/15175882/48885641-e1ffe780-eddd-11e8-9bdb-fe2a136be585.png)

here is without year
![accuracy_without_year](https://user-images.githubusercontent.com/15175882/48885654-e9bf8c00-eddd-11e8-9d99-3e0fd9fcfbcc.png)


I guess the reason for this is that if we removed the most deciding factors our model will do poor job. which also make sense...

**Comment by Mikeymice** [comment]
it may have something to do with number of stations as we discussed before. for the sake of our project. I guess we need to find a way to interpret it or have a way to explain it

**Comment by Mikeymice** [comment]
Included 'year' in features
